Bio/SeqIO/_index.py

```diff
@@ -32,37 +32,24 @@
 from io import StringIO
 
 from Bio import SeqIO
-from Bio import Alphabet
 from Bio.File import _IndexedSeqFileProxy, _open_for_random_access
 
 
 class SeqFileRandomAccess(_IndexedSeqFileProxy):
     """Base class for defining random access to sequence files."""
 
-    def __init__(self, filename, format, alphabet):
+    def __init__(self, filename, format):
         """Initialize the class."""
         self._handle = _open_for_random_access(filename)
-        self._alphabet = alphabet
         self._format = format
         # Load the parser class/function once an avoid the dict lookup in each
         # __getitem__ call:
         i = SeqIO._FormatToIterator[format]
-        # The following alphabet code is a bit nasty... duplicates logic in
-        # Bio.SeqIO.parse()
-        if alphabet is None:
-
-            def _parse(handle):
-                """Dynamically generated parser function (PRIVATE)."""
-                return next(i(handle))
-
-        else:
-            # TODO - Detect alphabet support ONCE at __init__
-            def _parse(handle):
-                """Dynamically generated parser function (PRIVATE)."""
-                try:
-                    return next(i(handle, alphabet=alphabet))
-                except TypeError:
-                    return next(SeqIO._force_alphabet(i(handle), alphabet))
+
+        # TODO - can we simplify this now?
+        def _parse(handle):
+            """Dynamically generated parser function (PRIVATE)."""
+            return next(i(handle))
```

Comment:
What does this do?

Reply:
It was the simplest way to remove the if/else, the TODO commend was a reminder to revisit this later.

Reply:
You could just store `i` and use that directly.

Reply:
Testing a cleaned up version now.